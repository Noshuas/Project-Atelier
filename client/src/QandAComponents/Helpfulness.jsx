import React, {useState, useEffect} from 'react';
import QAapiCalls from './QandAAPIcalls.js';

function Helpfulness (props) {

  let [feedback, setFeedback] = useState(false);

  let handleClick = (event) => {
    event.preventDefault();
    QAapiCalls.postHelpfullnessFeedback(props.QorA, props.id, 'helpful')
      .then( (res) => {
        setFeedback(true);
      })
      .catch( (err) => { console.log('err:', err); });
  };
  if (!feedback) {

    return (
      <>
        <span> Helpful? <button className="link-style-btn" onClick={handleClick}>Yes</button> ({props.helpfulness})
        </span>
      </>
    );
  }
  return (
    <>
      <span> Helpful?  Yes ({props.helpfulness + 1})  </span>
    </>
  );
}

export default Helpfulness;