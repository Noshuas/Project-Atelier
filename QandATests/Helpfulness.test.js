/**
 * @jest-environment jsdom
 */
 import React, { useState } from 'react';
 import renderer from 'react-test-renderer';
 import Helpfulness from '../client/src/QandAComponents/Helpfulness.jsx';
 import ReactDOM from 'react-dom';
 import Helpers from '../client/src/QandAComponents/helpers.js';
 import { render, screen } from '@testing-library/react';

 it('renders without crashing', () => {
   const div = document.createElement("div");
   ReactDOM.render(<Helpfulness helpfulness='7' QorA='questions' id='17076' />, div);
 });

 test('Helpfulness snapshot', () => {
   const component = renderer.create(
     <Helpfulness helpfulness='7' QorA='questions' id='17076' />
   );
   let tree = component.toJSON();
   expect(tree).toMatchSnapshot();
 });

 test('Helpfulness takes the helpfulness prop and displays it correctly', () => {
   const component = renderer.create(
     <Helpfulness helpfulness='7' QorA='questions' id='17076' />
   );

   let tree = component.toJSON();
   let spanText = Helpers.testHelp(tree.children);
   expect(spanText).toBe(' Helpful? Yes (7)');

   const component2 = renderer.create(
     <Helpfulness helpfulness='0' QorA='questions' id='17076' />
   );
   tree = component2.toJSON();
   spanText = Helpers.testHelp(tree.children);
   expect(spanText).toBe(' Helpful? Yes (0)');

   const component3 = renderer.create(
     <Helpfulness helpfulness='####' QorA='questions' id='17076' />
   );
   tree = component3.toJSON();
   spanText = Helpers.testHelp(tree.children);
   expect(spanText).toBe(' Helpful? Yes (####)');
 })