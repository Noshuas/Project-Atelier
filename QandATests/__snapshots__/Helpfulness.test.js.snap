// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpfulness snapshot 1`] = `
<span>
   Helpful? 
  <a
    href="#"
    onClick={[Function]}
  >
    Yes
  </a>
   (
  7
  )
</span>
`;
